app/models/concerns/orchestration/compute.rb

```diff
@@ -207,14 +207,20 @@ def find_image
   end
 
   def validate_compute_provisioning
-    return true unless image_build?
-    return true if ( compute_attributes.nil? or (compute_attributes[:image_id] || compute_attributes[:image_ref]).blank? )
-    img = find_image
-    if img
-      self.image = img
+    return true if compute_attributes.nil?
+    if image_build?
+      return true if (compute_attributes[:image_id] || compute_attributes[:image_ref]).blank?
+      img = find_image
+      if img
+        self.image = img
+      else
+        failure(_("Selected image does not belong to %s") % compute_resource) and return false
+      end
     else
-      failure(_("Selected image does not belong to %s") % compute_resource) and return false
+      # don't send the image information to the compute resource unless using the image provisioning method
+      [:image_id, :image_ref].each { |image_key| compute_attributes.delete(image_key) }
     end
+    true
```

Comment:
Omitting the fact that the exit value of the orchestration methods has it's meeting bit us many times. I don't think that relying that neither of the conditional branches  will not have false as exit value (such as `logger.info` when level is set to `warn`: just an example) is a good idea.